### PR TITLE
Generate: add missing logits processors docs

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

Some logits processor classes were missing on the docs. This PR corrects it by:
1 - Adding all missing logits processors to the docs
2 - Adding missing top-level imports (to stay consistent with previously existing logits processors)

### CODE REVIEW

**Comment by gante** [comment]
I've confirmed that the previously missing docs are now properly rendered in the doc preview 👍

File: docs/source/en/internal/generation_utils.md
Comment (ArthurZucker) [inline docs/source/en/internal/generation_utils.md:1]: Is there a logic behind the class order?

**Review by ArthurZucker — COMMENTED** [review]
Would be cool if the different logits appear in the small summary to directly jump to one or another: 
image
given that there are so many different ones WDYT? 

Also I noticed that the `logit_process.py` is not doctested, might be because some of the examples use big models, but would be cool to support them.

File: docs/source/en/internal/generation_utils.md
Comment (gante) [reply docs/source/en/internal/generation_utils.md:1]: None, they are in the same order as in the code. Happy to reorder both (classes in .py files and autodoc entries) in alphabetical order, to stay consistent with the imports, if you also agree!

**Comment by gante** [comment]
@ArthurZucker re tests -- fully on board. If you're okay with it, I'd like to do it on a separate PR :)

File: docs/source/en/internal/generation_utils.md
Comment (ArthurZucker) [reply docs/source/en/internal/generation_utils.md:1]: Sure!

**Comment by gante** [comment]
@ArthurZucker ready for a re-review! 

To recap, the previously reviewed version:
- Added missing logits processor docs

This version:
- Updates the left-handed TOC for this page (splits the classes by framework, so it's easier to navigate)
- Sorts the logits processors in the docs by alphabetical order
- Adds missing output classes to the output class docs section

In parallel, other concerns raised in the PR are being tackled in another places:
- #25692 Adds logits processors to the doctests

**Comment by gante** [comment]
(preview of the left-handed side TOC)
Screenshot 2023-08-24 at 14 21 45

**Review by ArthurZucker — APPROVED** [review]
Great work 🤗 looks a lot better
